metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertCommentCodeAction.scala

```diff
@@ -0,0 +1,170 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.codeactions.CodeAction
+import scala.meta.internal.parsing.Trees
+import scala.meta.pc.CancelToken
+import scala.meta.tokens.Token
+import scala.meta.tokens.Tokens
+
+import org.eclipse.{lsp4j => l}
+import scala.meta.io.AbsolutePath
+
+class ConvertCommentCodeAction(buffers: Buffers) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+
+    (for {
+      content <- buffers.get(path)
+      tokens <- tokenizeIfNotRangeSelection(range, content)
+      codeAction <- createActionIfPossible(
+        path,
+        tokens,
+        range,
+        isSingleLineComment(content),
+      )
+    } yield codeAction).toList
+  }
+
+  private def tokenizeIfNotRangeSelection(range: l.Range, content: String) = {
+    Option
+      .when(range.getStart == range.getEnd)(
+        Trees.defaultTokenizerDialect(content).tokenize
+      )
+      .flatMap(_.toOption)
+  }
+
+  private def isSingleLineComment(content: String)(
+      t: Token
+  ): Boolean = t match {
+    case tc: Token.Comment =>
+      val currentLine = content.split('\n')(t.pos.startLine)
```

Comment:
There were some fixes made recently in Metals and Scalameta regarding the handling of different EOL (end of line) characters, other than the `\n` (like `\r`, `\r\n`). Could there be a similar problem hidden here? :slightly_smiling_face: (CC @kasiaMarek)

Reply:
I think you might be right. I changed that to use `System.lineSeparator()`

Reply:
Best if you're maximally lenient about this and allow all `\n`, `\r\n` and `\r` to be new line no matter the system. This is more about how VS Code and other editors count lines than the system really, since all modern ones end with `\n` (either `\n` or '\r\n`).  Here's a discussion on carriage return in scalameta: https://github.com/scalameta/scalameta/pull/3343.

Now the tests on Windows fail, 'cause I think in the tests we use `\n`.